source/js/buyers-guide/components/creepometer/creepometer.jsx

```diff
@@ -4,12 +4,89 @@ export default class Creepometer extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {};
+    this.state = {
+      isHandleGrabbed: false,
+      handleOffset: 0
+    };
+
+    this.handleWidth = 70; // px
+    this.faceCount = 40; // Number of face frames
+    this.encodedStepCount = 100; // Upper range of values to be recorded
+    this.framePath = `/_images/buyers-guide/faces/`;
+
+    this.slideStart = this.slideStart.bind(this);
+    this.slideMove = this.slideMove.bind(this);
+    this.slideStop = this.slideStop.bind(this);
+
+    this.setSliderRef = element => {
+      this.sliderElement = element;
+    };
+  }
+
+  componentDidMount() {
+    // Slight delay because Firefox is too dang fast
```

Comment:
!

Reply:
I realize this is a *tad* hacky, but otherwise you'll need to write a bunch of custom DOM measurement code to verify that the layout is ready for measurement, which could get very tricky.